backend/src/hooks/check-has-permission/check-has-permission.hook.ts

```diff
@@ -0,0 +1,34 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+
+import { HttpCode, PermissionKey } from '~/common/enums/enums';
+import { UsersByIdResponseDto } from '~/common/types/types';
+import { PermissionsError } from '~/exceptions/exceptions';
+
+const checkHasPermissions =
+  (...permissions: PermissionKey[]) =>
+  async (req: FastifyRequest, rep: FastifyReply): Promise<void> => {
+    try {
+      const userPermissions =
+        (
+          req.user as UsersByIdResponseDto & {
+            permissions: [];
+          }
```

Comment:
you should modify type here `backend/src/index.d.ts`

Reply:
What if I need this type on frontend too, until user doesn't have permission field?

Reply:
not we have functionality with user permissions